### ISSUE TITLE
[docs] `select` aliasing not documented

### ISSUE THREAD

**Issue by lourd** [tgriesser/knex#1178]
As seen in one of @tgriesser's presentations from last year https://speakerdeck.com/tgriesser/making-relational-cool-again-or-javascript-on-acid, see slide 50.

Discussed a little on [this issue](https://github.com/tgriesser/knex/issues/139)

**Comment by jmm** [comment]
Agreed, this should be documented. Per the concerns in https://github.com/tgriesser/knex/issues/139 about simply splitting on `as`, perhaps the relevant methods should accept 2-element arrays, example:

```js
.select(['real_name', 'alias'])
```

Or require identifiers with special characters to be quoted, example:

```js
.select('"smart as a fox"')
```

In any case, whatever the implementation, it should be documented.

**Comment by mercmobily** [comment]
This issue is no longer a problem! The documentation for select aliasing has actually been added since this issue was created in 2016. 

Looking at the current documentation in `docs/src/guide/query-builder.md:9-11`, there's now a dedicated "Identifier Syntax" section that clearly explains both aliasing methods:

1. String syntax: `'columnName as aliasName'` 
2. Object syntax: `{ aliasName: 'columnName' }`

The documentation also includes multiple examples throughout showing both approaches:

```javascript
// String syntax examples
knex.select('id as identifier').from('users');
knex.select('cities.name as cityName', 'country.name as countryName');

// Object syntax examples  
knex.column('title', { by: 'author' }, 'year').select().from('books');
knex.select({ aTitle: 'a.title', bTitle: 'b.title' });
```

The `.column()` method documentation specifically mentions: "Passing an object will automatically alias the columns with the given keys."

However, there is a related ongoing issue #6023 about the problems with how Knex handles the "as" keyword in column names (it splits on "as" even when "as" is part of the actual column name). This is the real issue that needs addressing - the aliasing functionality works, but the string parsing can be problematic.

This original documentation issue can be closed since the requested documentation now exists and is comprehensive.

**Related issues:**
- #6023 - Current issue with "as" splitting in column names  
- #5382 - Top level aliasing support request for RedShift
- #139 (referenced in original issue) - Historical discussion about "as" splitting concerns
